src/transformers/integrations/deepspeed.py

```diff
@@ -282,9 +283,20 @@ def unset_hf_deepspeed_config():
     _hf_deepspeed_config_weak_ref = None
 
 
-def is_deepspeed_zero3_enabled():
+def is_deepspeed_zero3_enabled(check_accelerate=False):
+    """
+    If `check_accelerate`, will also check if `deepspeed_zero3` has been enabled through
+    the `HfDeepSpeedConfig` and if it was enabled through the environment variables
+    from `accelerate launch`.
+    """
+    accelerate_zero_stage = int(os.environ.get("ACCELERATE_DEEPSPEED_ZERO_STAGE", -1))
+    accelerate_zero_init = os.environ.get("ACCELERATE_DEEPSPEED_ZERO3_INIT", "0")
     if _hf_deepspeed_config_weak_ref is not None and _hf_deepspeed_config_weak_ref() is not None:
         return _hf_deepspeed_config_weak_ref().is_zero3()
+    # This only gets triggered passively if the user launches code with a configured
+    # `accelerate launch` without making `TrainingArguments`
+    elif check_accelerate and accelerate_zero_stage != -1 and accelerate_zero_init != "0":
+        return True, False
```

Comment:
The bad news with this is it's a conditional return, which I'm not a fan of, but this has to check both setups in this very particular case.

Reply:
Yes, I don't think this ideal. When I call `is_deepspeed_zero3_enabled` I expect a boolean return. Given that we already return on L294, passing `check_accelerate` isn't guaranteed to return two values. 

What I would suggest is creating a new method e.g. `check_zero_init_init` or `check_deepspeed_accelerate_enabled` which does the accelerate checks and can be called in `modeling_utils`

Reply:
Done, I've made `is_deepspeed_zero3_enabled_accelerate`, since we don't need anything more complex than that imo